KarhooSDK/Network/HttpClient/TokenRefreshingHttpClient/Interactor/KarhooRefreshTokenInteractor.swift

```diff
@@ -140,10 +161,29 @@ final class KarhooRefreshTokenInteractor: RefreshTokenInteractor {
             return true
         }
         let timeToExpiration = date.timeIntervalSince1970 - Date().timeIntervalSince1970
-        return timeToExpiration < Constants.MaxTimeIntervalToRefreshToken
+        return timeToExpiration < Constants.refreshBuffer
     }
 
-    private struct Constants {
-        static let MaxTimeIntervalToRefreshToken = TimeInterval(30) // 30 sec
+    private func scheduleRefreshTokenTimer() {
+        guard let credentials = dataStore.getCurrentCredentials() else {
+            assertionFailure("Credentials should be set at this stage")
+            return
+        }
+        refreshTokenTimer?.invalidate()
+        refreshTokenTimer = nil
+
+        let secondsToRefresh: TimeInterval = max(0, (credentials.expiryDate?.timeIntervalSinceNow ?? 0) - Constants.refreshBuffer)
+
+        refreshTokenTimer = Timer.scheduledTimer(
+            withTimeInterval: secondsToRefresh,
+            repeats: false,
+            block: { [weak self] _ in
+                self?.refreshToken { result in
```

Comment:
we also need a timer inside  the `requestExteralAuthentication()` method. If the DP takes more than 60 seconds to get back to us then we need to return an error
cc: @mateid-kr

Reply:
Just realised there's a type in the method name too